### PR TITLE
 OIA-52: Implement deletion of SCV entries 

### PR DESCRIPTION
**Pull request by fooker**
Depends on https://github.com/OpenNMS/opennms-integration-api/pull/76

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/OIA-52

### CODE REVIEW

**Review by RangerRick — DISMISSED** [review]
Looks reasonable just make sure the OPA side of this is merged and you relaunch the build so it goes green before we merge this.

File: features/api-layer/common/src/main/java/org/opennms/features/apilayer/common/scv/SecureCredentialsVaultImpl.java [hunk 1]
```diff
@@ -76,4 +76,10 @@ public void setCredentials(String alias, Credentials credentials) {
                 credentials.getAttributes())
         );
     }
+
+    @Override
+    public void deteleCredentials(String alias) {
```
Comment (synqotik) [inline features/api-layer/common/src/main/java/org/opennms/features/apilayer/common/scv/SecureCredentialsVaultImpl.java:81]: Noticed this typo, `deteleCredentials` -> `deleteCredentials`

**Review by aramosvizcarra — DISMISSED** [review]
Looks good to me

File: features/scv/scvcli/src/main/java/org/opennms/features/scv/cli/commands/DeleteCommand.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2017 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2017 The OpenNMS Group, Inc.
```
Comment (christianpape) [inline features/scv/scvcli/src/main/java/org/opennms/features/scv/cli/commands/DeleteCommand.java:5]:
```suggestion
 * Copyright (C) 2022 The OpenNMS Group, Inc.
 * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
```

File: features/scv/shell/src/main/java/org/opennms/features/scv/shell/ScvDeleteCommand.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2016 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2016 The OpenNMS Group, Inc.
```
Comment (christianpape) [inline features/scv/shell/src/main/java/org/opennms/features/scv/shell/ScvDeleteCommand.java:5]:
```suggestion
 * Copyright (C) 2022 The OpenNMS Group, Inc.
 * OpenNMS(R) is Copyright (C) 1999-2022 The OpenNMS Group, Inc.
```

File: features/scv/shell/src/main/java/org/opennms/features/scv/shell/ScvDeleteCommand.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import org.opennms.features.scv.api.SecureCredentialsVault;
+
+@Command(scope = "opennms", name = "scv-delete", description="Delete the stored credentials for the given alias.")
+@Service
+public class ScvDeleteCommand implements Action {
+
+    @Reference
+    public SecureCredentialsVault secureCredentialsVault;
+
+    @Argument(index = 0, name = "alias", description = "Alias used to retrieve the credentials.", required = true, multiValued = false)
```
Comment (christianpape) [inline features/scv/shell/src/main/java/org/opennms/features/scv/shell/ScvDeleteCommand.java:45]:
```suggestion
    @Argument(index = 0, name = "alias", description = "Alias for the credentials to be deleted.", required = true, multiValued = false)
```

**Review by christianpape — CHANGES_REQUESTED** [review]
Some minor things to do...

**Review by christianpape — DISMISSED** [review]
LGTM.

**Pull request merged**